File: check-items-unsorted.js
#!/usr/bin/env node

const Dex = require('./../pokemon-showdown-ts/dist/sim/dex').Dex;

const dex = Dex.mod('gen9');

// Get items WITHOUT sorting (as used in generate-teams.js)
const allItems = Object.values(dex.items.all()).map(i => i.id);

console.log('Total items:', allItems.length);
console.log('First 20 items (unsorted):', allItems.slice(0, 20));
console.log('');

// Find specific items
console.log('Index of firiumz:', allItems.indexOf('firiumz'));
console.log('Index of fightiniumz:', allItems.indexOf('fightiniumz'));

File: examples/check_items_unsorted.rs
/// Check unsorted items order in Rust

use pokemon_showdown::Dex;

fn main() {
    let dex = Dex::load_default().expect("Failed to load dex");

    // Get items WITHOUT sorting
    let all_items: Vec<_> = dex.items.keys().map(|id| id.as_str()).collect();

    println!("Total items: {}", all_items.len());
    println!("First 20 items (unsorted): {:?}", &all_items[0..20.min(all_items.len())]);
    println!();

    // Find specific items
    println!("Index of firiumz: {:?}", all_items.iter().position(|&x| x == "firiumz"));
    println!("Index of fightiniumz: {:?}", all_items.iter().position(|&x| x == "fightiniumz"));
}
